spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb

```diff
@@ -52,13 +52,56 @@
     end
   end
 
+  context "an ActiveModel with a validation on an attribute that delegates to a has_many association in another ActiveModel" do
+    before do
+      parent = define_active_model_class('Parent', accessors: [:children]).new
+      @model = define_active_model_class('Example', accessors: [:parent]) do
+        delegate :children, :children=, to: :parent
+        validates_presence_of :children
+      end
+      .new(parent: parent)
+    end
+
+    it 'accepts' do
+      expect(@model).to validate_presence_of(:children)
+    end
+  end
+
+  context "an ActiveModel with a validation on an attribute that delegates to a habtm association in another ActiveModel" do
+    before do
+      define_model :child
+      parent = define_model :parent do
+        has_and_belongs_to_many :children
+        validates_presence_of :children
+      end
+      .new
+
+      create_table 'children_parents', id: false do |t|
+        t.integer :child_id
+        t.integer :parent_id
+      end
+
+      @model = define_active_model_class('Example', accessors: [:parent]) do
+        delegate :children, :children=, to: :parent
+        validates_presence_of :children
+      end
+      .new(parent: parent)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 0) spaces for indenting an expression in an assignment spanning multiple lines.